### PR TITLE
Brass monkey

### PR DESCRIPTION
**Pull request by ironsnake345**
## About The Pull Request
This is a redux of my old brass monkey addition, since TG's code had some fuckery that moved it to its new repository. This PR does two things: adds Brass Monkey, a drink crafted by mixing Kong with iron, copper and banana juice, and updates monkey aggression code to check for mobs with the Jungle faction as non-eligible targets for monkey aggression, instead of just checking for other monkeys as non-eligible. Now you have a highly alcoholic drink that makes monkeys not wanna kill you, and monkeys won't attack other jungle mobs. Yay!

## Why It's Good For The Game
It's a silly addition, which also causes monkey's aggression code to make a bit more thematic sense. Who doesn't love silly, memey, far-flung jokes about very old and criminally catchy pop songs? Besides, spess is a scary place. A good spessman always needs a new way to get drunk off his ass.

## Changelog
:cl:
add: Added Brass Monkey, an alcoholic drink mixed by combining 5u Kong, 5u Banana juice, 1u copper, and 1u iron. It makes 10 units of a drink that makes monkeys very hesitant to try and kill you while it's in your system! Who could assault such a funky monkey as you?
tweak: Monkeys are now jungle mobs
balance: Monkeys are now jungle mobs
imageadd: added a sprite for a glass of Brass Monkey
code: Monkey aggression code now decides whether or not they want to attack something based off of whether or not it is a jungle mob or has the Monkeybro trait, not whether or not it's another monkey.
/:cl:

### CODE REVIEW

**Comment by ATH1909** [comment]
Is it intentional that people without livers can retain the effects of this chem indefinitely?

Did the microdosing bugs related to on_mob_delete() get fixed? If not, you might want to try using on_mob_metabolize() and on_mob_stop_metabolize() instead of on_mob_add() and on_mob_delete().

**Comment by ironsnake345** [comment]
Good idea, ATH! I hadn't considered that possibility. Plus, I'm pretty new to the whole TG coding thing tbh, so advice like that is welcome.

File: code/modules/mob/living/carbon/monkey/combat.dm [hunk 1]
```diff
@@ -123,8 +123,8 @@
 	if(enemies[L])
 		return TRUE
 
-	// target non-monkey mobs when aggressive, with a small probability of monkey v monkey
-	if(aggressive && (!istype(L, /mob/living/carbon/monkey/) || prob(MONKEY_AGGRESSIVE_MVM_PROB)))
+	// target non-jungle mobs when aggressive, with a small probability of monkey v monkey. Also avoids attacking  monkey bros.
+	if(aggressive && (!faction_check_mob(L, FALSE) || !HAS_TRAIT(L, TRAIT_MONKEYBRO) || prob(MONKEY_AGGRESSIVE_MVM_PROB)))
```
Comment (ATH1909) [inline code/modules/mob/living/carbon/monkey/combat.dm:127]:
```suggestion
	if(aggressive && (!(faction_check_mob(L, FALSE) || HAS_TRAIT(L, TRAIT_MONKEYBRO)) || prob(MONKEY_AGGRESSIVE_MVM_PROB)))
```

**Comment by Anonmare** [comment]
Cold enough to freeze the balls off a brass monkey

File: code/modules/reagents/chemistry/reagents/alcohol_reagents.dm [hunk 2]
```diff
@@ -2239,3 +2239,21 @@ All effects don't start immediately, but rather get worse over time; the rate is
...
+	color = "#e9e357"
+	description = "An ancient beverage supposedly invented by a bunch of monkeys trying to join a failing steampunk cult. Traditionally served in a clockwork mug."
+	boozepwr = 85
+	taste_description = "steam-powered bananas"
+	glass_icon_state = "brass_monkey"
+	glass_name = "Brass Monkey"
+	glass_desc = "Coolin' by the lockers, getting kind of funky, me and the crew, we're drinking Brass Monkey."
+
+/datum/reagent/consumable/ethanol/brass_monkey/on_mob_metabolize(mob/living/M)
+	if(!HAS_TRAIT_FROM(M, TRAIT_MONKEYBRO, BRASS_MONKEY))
```
Comment (Qustinnus) [inline code/modules/reagents/chemistry/reagents/alcohol_reagents.dm:2254]: i dont think you have to check this it does this on its own but dont quote me on that

File: code/modules/reagents/chemistry/reagents/alcohol_reagents.dm [hunk 3]
```diff
@@ -2239,3 +2239,21 @@ All effects don't start immediately, but rather get worse over time; the rate is
...
+		ADD_TRAIT(M, TRAIT_MONKEYBRO, BRASS_MONKEY)
+
+/datum/reagent/consumable/ethanol/brass_monkey/on_mob_end_metabolize(mob/living/M)
+	if(HAS_TRAIT_FROM(M, TRAIT_MONKEYBRO, BRASS_MONKEY))
```
Comment (Qustinnus) [inline code/modules/reagents/chemistry/reagents/alcohol_reagents.dm:2258]: ditto

Comment (ironsnake345) [reply code/modules/reagents/chemistry/reagents/alcohol_reagents.dm:2258]: You're probably right, but I want to go on the safe side of things. Besides, I seriously doubt one HAS_TRAIT_FROM proc whenever someone sips this drink will have a noticeable effect on performance.

File: code/modules/mob/living/carbon/monkey/combat.dm (hunk 1 above)
Comment (ironsnake345) [reply code/modules/mob/living/carbon/monkey/combat.dm:127]: Oh, good idea, I like that. Committing!

File: code/modules/reagents/chemistry/reagents/alcohol_reagents.dm (hunk 3 above)
Comment (Qustinnus) [reply code/modules/reagents/chemistry/reagents/alcohol_reagents.dm:2258]: that's not a good argument. on its own barely any check will really dent performance on its own.  It's the accumulative dent tons of those checks would have if noone cared that would affect performance, though.

@Rohesie you worked with the trait code, does it matter if you remove / add traits without checking first? I'd imagine not right?

Comment (Rohesie) [reply code/modules/reagents/chemistry/reagents/alcohol_reagents.dm:2258]: This is very much a no.
This check is superfluous at best and error-hiding at worst.
Use `ADD_TRAIT(M, TRAIT_MONKEYBRO, type)` and `REMOVE_TRAIT(M, TRAIT_MONKEYBRO, type)`
No need to define a new string for a reagent, the type is the convention.

**Comment by ExcessiveUseOfCobblestone** [comment]
just re-pr when you do reviews and fix conflicts.

**Pull request closed without merging**